### DISCUSSION TITLE
Is it possible to see stdout logs of the application?

### DISCUSSION THREAD

**Discussion by mucahitkantepe** [sweetpad-dev/sweetpad#2] [Any discussions (ideas, questions, ...)]
Hi,

Thanks for taking the initiative to fill the missing piece in the VsCode ecosystem! 

My app has some print commands here and there to debug but I could not find where I can see them, is it not implemented or am I missing something?

**Comment by hyzyla** [comment]
Hello! Thank you for the kind words. I plan to add stdout/stderr output directly in the console. I don't know when yet, as I am working on the extension in my free time, but I definitely have it in my plans. I can ping you here, when it will be ready

How does the extension work for you in general? Is there anything that doesn't work or anything else that should be added?

**Reply by hyzyla** [reply]
Oh, it was easier than I thougth 😄 Added to 0.1.1 version

![CleanShot 2024-03-03 at 19 37 43@2x](https://github.com/sweetpad-dev/sweetpad/assets/7952949/1292d223-bbb4-4032-94ce-bc0a6e003535)

**Reply by mucahitkantepe** [reply]
I am not a very pro user or a pro ios dev but I think it is quite simple and works well, I didn't really like coding on XCode, so this is just what I needed. I can create issues as I hit bugs or miss anything!

**Reply by cameroncooke** [reply]
This is great but it doesn't appear to work for macOS apps?

**Reply by kenryu42** [reply]
Yes, I can confirm the stdout/stderr output is not working for macOS apps.

**Reply by FelixLisczyk** [reply]
The console output for macOS apps is only visible after the running app has been terminated. I have submitted a feature request to enhance the logging integration: https://github.com/sweetpad-dev/sweetpad/issues/97

**Reply by dmik** [reply]
JFYI, if your only concern is to instantly see the app output in the internal VS Code terminal where SweetPad runs it, doing this in your `settings.json` is sufficient:
```json
    "sweetpad.build.launchEnv": {
        "NSUnbufferedIO": "YES"
    },
 ```
 
 As a bonus, doing a similar setup for the build environment makes `xcodebuild` output also instant even when run through `xcbeautify` (a similar problem due to buffered I/O when using pipes for redirection in SweetPad I suppose):
```json
    "sweetpad.build.env": {
        "NSUnbufferedIO": "YES"
    },
```
